src/api/searchData.ts

```diff
@@ -0,0 +1,36 @@
+import type { FormEvent } from 'react';
+import type { ICountry } from '../types/tcard';
+export function searchData(evt: FormEvent, arrForSearch: ICountry[]) {
+  const target = evt.currentTarget;
+
+  if (target instanceof HTMLFormElement) {
```

Comment:
```suggestion
interface Props {
  event: FormEvent<HTMLFormElement>,
  arrForSearch: ICountry[]
}

export function searchData({evt, arrForSearch}: Props) {
  const target = evt.currentTarget;

```
там ниже нужно будет удалять ненужную "}" просто пример ка правильно типизировать что бы не пришлось лишний раз проверять instance